lib/collection/src/collection/mod.rs

```diff
@@ -782,6 +795,39 @@ impl Collection {
     pub async fn trigger_optimizers(&self) {
         self.shards_holder.read().await.trigger_optimizers().await;
     }
+
+    async fn load_segment_stats(shards_holder: &Arc<RwLock<ShardHolder>>) -> LocalDataStats {
+        let shard_lock = shards_holder.read().await;
+        shard_lock.load_local_segments_stats().await
+    }
+
+    /// Checks and performs a cache update for local data statistics if needed.
+    /// Returns `Some(..)` with the new values if a cache update has been performed and `None` otherwise.
+    async fn check_and_update_local_size_stats(&self) -> Option<LocalDataStats> {
+        if self.local_stats_cache.check_need_update_and_increment() {
+            let new_stats = Self::load_segment_stats(&self.shards_holder).await;
+            self.local_stats_cache.update(new_stats);
+            return Some(new_stats);
+        }
+
+        None
+    }
+
+    /// Returns the local data statistics for this collection, cached and auto-updated.
+    pub async fn local_data_stats(&self) -> LocalDataStats {
+        if let Some(shard_stats) = self.check_and_update_local_size_stats().await {
+            return shard_stats;
+        }
+        self.local_stats_cache.get()
+    }
+
+    /// Returns the estimated local vector storage size for this collection, cached and auto-updated.
+    pub async fn estimated_local_vector_storage_size(&self) -> usize {
+        if let Some(shard_stats) = self.check_and_update_local_size_stats().await {
+            return shard_stats.vector_storage_size;
+        }
+        self.local_stats_cache.get_vector_storage()
+    }
```

Comment:
We don't check both in a request. The first function was intended for places where we need all data eg. if we want to sync those stats with other nodes in a cluster or check all statistics in a request and is currently not in use.
The second function only returns vector storage size to prevent fetching more atomic values than we actually need for less performance impact.

If you still think we should remove it and add when needed, I'll remove it from this PR.

Reply:
If you're not 99% but 100% sure you'll use it later, then I'm happy :smiley: